satellite/metainfo/validation.go

```diff
@@ -260,9 +260,20 @@ func (endpoint *Endpoint) validatePointer(ctx context.Context, pointer *pb.Point
 func (endpoint *Endpoint) validateRedundancy(ctx context.Context, redundancy *pb.RedundancyScheme) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	// TODO more validation, use validation from eestream.NewRedundancyStrategy
-	if redundancy.ErasureShareSize <= 0 {
-		return Error.New("erasure share size cannot be less than 0")
+	if endpoint.rsConfig.Validate == true {
+		if redundancy.ErasureShareSize <= 0 ||
+			redundancy.MinReq >= redundancy.RepairThreshold {
+			return Error.New("invalid redundancy parameters")
+		}
+
+		if endpoint.rsConfig.MaxThreshold == int(redundancy.Total) &&
```

Comment:
make this logic more like `endpoint.rsConfig.MaxThreshold != int(redundancy.Total) || ...` and if `true` return error. Now it needs to check all `if` conditions to know if `rs` doesn't fit configured.

Reply:
updated